src/game/crossteaser/mod.rs

```diff
@@ -53,31 +53,586 @@ const ABOUT: &str = "PLACEHOLDER";
+            front: o.front,
+            top: o.top,
+            right: o.right,
+        };
+        while t_list & 0b11 != 0 {
+            transform = t_list & 0b11;
+            t_list >>= 2;
+            axis = t_list & 0b111;
+            t_list >>= 3;
+            if transform == 0b01 {
+                new_o = mov::cw_on_axis(&new_o, axis);
+            } else if transform == 0b10 {
+                new_o = mov::axis_180(&new_o, axis);
+            } else if transform == 0b11 {
+                new_o = mov::cw_on_axis(&new_o, 5 - axis);
+            }
+        }
+        return new_o;
+    }
+
+    /// Reduces the state to one with last piece orientation of 0
+    /// and adjusting the rest of the pieces equivalently
+    /// Note that this does not mean applying the exact same transformation
+    /// to every piece.
+    /// See apply_transformation() for additional details.
+    fn reduce(&self, s: &UnhashedState) -> UnhashedState {
+        let mut new_pieces: Vec<Orientation> = Vec::new();
+        let pos: u64 = hash_orientation(&s.pieces.last().unwrap());
+        if pos == 0 {
+            return s.deep_copy();
+        }
+        let transform_list: u64 = TRANSFORM_MAP[pos as usize];
+        for o in &s.pieces {
+            new_pieces.push(self.apply_transformations(o, transform_list));
+        }
+        return UnhashedState {
+            pieces: new_pieces,
+            free: s.free,
+        };
+    }
+
+    /// Applies 4 physical board symmetries and finds the canonical of the set
+    /// -Rotate 180
+    /// -Flip board & rotate 90
+    /// -Flip board & rotate 270
+    /// NOTE: for boards with unequal dimensions, cannot apply
+    /// Flip board & rotate 90 or flip board & rotate 270 symmetries.
+    /// Not Currently in use.
+    fn board_sym(&self, s: &UnhashedState) -> UnhashedState {
+        let mut sym_list: Vec<UnhashedState> = Vec::new();
+        sym_list.push(s.deep_copy());
+        sym_list.push(self.board_180(s));
+        if self.width == self.length {
+            let flipped: UnhashedState = self.flip_90(s);
+            sym_list.push(flipped.deep_copy());
+            sym_list.push(self.board_180(&flipped));
+        }
+        let mut min: u64 = self.hash(&sym_list[0]);
+        let mut min_i: usize = 0;
+        for i in 0..sym_list.len() {
+            let curr: u64 = self.hash(&sym_list[i]);
+            if curr < min {
+                min_i = i;
+                min = curr;
+            }
+        }
+        return sym_list.remove(min_i);
+    }
+
+    /// Applies 4 board symmetries and then reduces state to canonical
+    fn canonical(&self, s: &UnhashedState) -> UnhashedState {
+        return self.reduce(&s);
+    }
+}
+
+/// Module which contains all transition helper functions
+/// There are 4 functions for rotating an individual piece which represents a
+/// shift in the given direction as defined by the structure of the crossteaser
+/// game board.
+mod mov {
```

Comment:
It's all good to keep all of these, but instead of making this a `mod mov`, let's make it an `impl Orientation`.